examples/1_advanced/8_status_list_2021.rs

```diff
@@ -0,0 +1,178 @@
+  // (actually it will stay in memory).
+  let status_list_credential = StatusList2021CredentialBuilder::new(status_list)
+    .purpose(StatusPurpose::Revocation)
+    .subject_id(Url::parse("http://example.com/credential/status")?)
+    .issuer(Issuer::Url(issuer_document.id().to_url().into()))
+    .build()?;
+
+  // Create a credential subject indicating the degree earned by Alice.
+  let subject: Subject = Subject::from_json_value(json!({
+    "id": alice_document.id().as_str(),
+    "name": "Alice",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts",
+    },
+    "GPA": "4.0",
+  }))?;
+
+  // Create an unsigned `UniversityDegree` credential for Alice.
+  // The issuer also chooses a unique `StatusList2021` index to be able to revoke it later.
+  let credential_index: usize = 420;
+  let status: Status = StatusList2021Entry::new(
+    status_list_credential.id().cloned().unwrap(),
+    status_list_credential.purpose(),
+    credential_index,
+    None,
+  )
+  .into();
+
+  // Build credential using subject above, status, and issuer.
+  let mut credential: Credential = CredentialBuilder::default()
+    .id(Url::parse("https://example.edu/credentials/3732")?)
+    .issuer(Url::parse(issuer_document.id().as_str())?)
+    .type_("UniversityDegreeCredential")
+    .status(status)
+    .subject(subject)
+    .build()?;
+
+  println!("Credential JSON > {credential:#}");
+
+  let credential_jwt: Jwt = issuer_document
+    .create_credential_jwt(
+      &credential,
+      &storage_issuer,
+      &fragment_issuer,
+      &JwsSignatureOptions::default(),
+      None,
+    )
+    .await?;
+
+  let validator: JwtCredentialValidator<EdDSAJwsVerifier> =
+    JwtCredentialValidator::with_signature_verifier(EdDSAJwsVerifier::default());
+  // The validator has no way of retriving the status list to check for the
+  // revocation of the credential. Let's skip that pass and perform the operation manually.
+  let mut validation_options = JwtCredentialValidationOptions::default();
+  validation_options.status = StatusCheck::SkipUnsupported;
+  // Validate the credential's signature using the issuer's DID Document.
+  validator.validate::<_, Object>(
+    &credential_jwt,
+    &issuer_document,
+    &validation_options,
+    FailFast::FirstError,
+  )?;
+
+  let status_list_credential_json = status_list_credential.to_json().unwrap();
+
+  // ===========================================================================
+  // Revocation of the Verifiable Credential.
+  // ===========================================================================
+
+  // The issuer retrieves the status list credential.
+  let mut status_list_credential =
+    serde_json::from_str::<StatusList2021Credential>(status_list_credential_json.as_str()).unwrap();
+
+  // Set the value of the chosen index entry to true to revoke the credential
+  status_list_credential.set_credential_status(&mut credential, credential_index, true)?;
```

Comment:
It's slightly more complex than that. A `StatusListCredential` has a purpose that cannot be changed after its creation. Therefore a `StatusListCredential` can only set or unset the status of a credential to its `statusPurpose`. If the `set_credential_status` method signature becomes:
```rust
fn set_credential_status(&mut self, c: &mut Credential, i: usize, status: CredentialStatus) -> Result<()>
```
we have lost some type correctness as we have to check that the passed `status` matches the `StatusListCredential`'s, thus the user is more prone to misusing the API.

Reply:
We could've combined them to `RevokedOrSuspended` since that represents the `1` value better and the purpose is not relevant anyways. I added a suggestion to change the `value` parameter to something more descriptive if you want to keep it the way it is.